### PR TITLE
Color in svg icons with text-color

### PR DESCRIPTION
**Pull request by robertgzr**
Took a shot at implementing #300. Not sure how to go about detecting if we should load it as a scalable so I match on the extension for now... This might need something more robust though.

### CODE REVIEW

File: icon.c [hunk 1]
```diff
@@ -18,14 +18,88 @@
...
+
+static GdkPixbuf *load_image(const char *path, struct mako_style *style) {
 	if (strlen(path) == 0) {
 		return NULL;
 	}
 	GError *err = NULL;
-	GdkPixbuf *pixbuf = gdk_pixbuf_new_from_file(path, &err);
+	GdkPixbuf *pixbuf;
+	// FIXME ok? or do we need to actually look at mime info?
+	if (strcmp(strrchr(path, '.'), ".svg") == 0) {
```
Comment (robertgzr) [inline icon.c:98]: not sure about this one... let me know if there is something better we can do? I've been using it with this and haven't had any issues so far.

**Comment by emersion** [comment]
I must say, I'm not a huge fan of mungling SVGs to set the icon colors… It seems like the classes are pretty GNOME-specific too?

**Comment by robertgzr** [comment]
@emersion yeah I basically followed https://gitlab.gnome.org/GNOME/gtk/-/blob/master/gtk/gdkpixbufutils.c#L335

I personally don't care about the classes, we can just drop those. `rect,circle,path` are basic svg elements so I don't see an issue with them. 

if we want to allow to match them with our foreground color I don't think there's a way around the munging...

**Comment by robertgzr** [comment]
I actually noticed that svg icons supplied by applications (e.g. mpv) weren't looking right previously...

now it will not overwrite icons that use a color fill already

**Comment by emersion** [comment]
Sorry, I'd rather not add this feature.

**Comment by octvs** [comment]
Is it possible to keep the feature living on a fork @robertgzr ? Analogous to [mortie/swaylock-effects](https://github.com/mortie/swaylock-effects) for [swaywm/swaylock](https://github.com/swaywm/swaylock).

**Pull request closed without merging**